src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -258,6 +277,36 @@ protected override void OnKeyUp(KeyEventArgs e)
             }
         }
 
+        private void SetComboboxCurrentVersion(string comboboxText, bool userTypedAVersionRange, IEnumerable<NuGetVersion> versions)
+        {
+            NuGetVersion matchVersion = null;
+            VersionRange userRange = null;
+
+            // Get the best version from the range if the user typed a custom version
+            bool userTypedAValidVersionRange = VersionRange.TryParse(comboboxText, out userRange);
+            if (userTypedAValidVersionRange && (userTypedAVersionRange || (userRange != null && userRange.IsFloating)))
+            {
+                matchVersion = userRange.FindBestMatch(versions);
+            }
+
+            // If the selected version is not the correct one, deselect a version so Install/Update button is disabled.
+            if (_versions.SelectedIndex != -1 &&
+                (_versions.Text != _versions.Items[_versions.SelectedIndex].ToString() || matchVersion?.ToString() != _versions.Items[_versions.SelectedIndex].ToString()))
```

Comment:
`_versions.Text != _versions.Items[_versions.SelectedIndex].ToString()`
Is this because the Text is bound to a different property than Items directly?

Reply:
Could make a variable for :` _versions.Items[_versions.SelectedIndex].ToString()`